src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +164,62 @@ class bo : public xrt::bo
   }
 };
 
+class profiling_impl;
+class profiling : public detail::pimpl<profiling_impl>
+{
+public:
+  enum class profiling_option : int { io_total_stream_running_to_idle_cycles = 0, io_stream_start_to_bytes_transferred_cycles = 1, io_stream_start_difference_cycles = 2, io_stream_running_event_count = 3 };
+
+  profiling() = default;
+
+  /**
+   * event() - Constructor from a device
+   *
+   * @param device
+   *  The device on which the profiling should start
+   *
+   */
+  explicit
+  profiling(const xrt::device& device);
+
+  /**
+   * start() - Start AIE performance profiling
+   *
+   * @param option
+   *  Profiling option
+   * @param port1_name
+   *  PLIO/GMIO port 1 name.
+   * @param port2_name
+   *  PLIO/GMIO port 2 name.
+   * @param value
```

Comment:
I know this is in some user guide. Without providing the hyperlink, you could reference UGxxxx for more detail. Without the UG, I would be baffled on the meaning of these parameters.